### PR TITLE
Issue/4571 simperium lag on comment pn open show pn payload

### LINKED ISSUE #4571: Notifications don’t always open on Android when tapping on a Push Notification
### Expected behavior
When tapping on a WordPress push notification in the device notification center, the full notification content should open inside the app.

### Actual behavior
I tap on the notification and the WordPress app opens on the notifications tab showing the full list of all my read notifications, but the one that I tapped on is not in the list, as if it doesn’t actually exist. 
(The toast "Could not open the notification" can appear for few secs).

If I then wait for a while (anything from a few seconds to a minute or two), the unread notification will appear at the top. 
Sometimes I have to kill the app completely and reopen it to get the notification to be displayed in the list.

### Steps to reproduce the behavior
This seems to happen mostly when I’m on a slower internet connection (mobile data with poor reception or unreliable wifi), so what I’m guessing is happening is that the push notification is received, but the synch with Simperium fails, or it's not started at all.
Btw, sometimes it happened also when I was on fast connections.

##### Tested on Nexus 5x Android 7.0

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #4571 

When simperium lags in syncing and we can't find the corresponding Note in the local bucket through its Note id,  this solution implements reading the Note from the Push Notification payload to be able to show it to the user.

To test: 

1. comment out the contents of method `startBuckets()`  in file `SimperiumUtils.java`. This will prevent Simperium from actually trying to sync buckets, so we make sure we are not getting info from Simperium as per the purpose of this test.
2. generate a comment on a blog you're an admin of (make the comment with another test user).
3. you should receive a push notification  (which internally carries the full not payload)
4. tap on such notification
5. the app will build the Note out of the PN payload and show the comment contents to the user.
6. also, the user should be able to moderate the comment accordingly

Needs review: @daniloercoli

### CODE REVIEW

**Comment by daniloercoli** [comment]
This should probably be merged in `6.0` instead of `develop` since it's a very annoying bug. We're calling users for an action (Sending them PNs but then nothing is available in the app).

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java [hunk 1]
```diff
@@ -167,13 +163,15 @@ private void buildAndShowNotificationFromNoteData(Bundle data) {
             Bundle noteBundle = sActiveNotificationsMap.get(id);
             if (noteBundle != null && noteBundle.getString(PUSH_ARG_NOTE_ID, "").equals(noteId)) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: I wonder why `note_id` is empty here. Aren't we always setting it on the server side?

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java [hunk 2]
```diff
@@ -193,11 +191,22 @@ private void buildAndShowNotificationFromNoteData(Bundle data) {
...
 
-    private void showGroupNotificationForActiveNotificationsMap(int pushId, String noteId, String noteType,
-                                                                String largeIconUri, String title, String message) {
+    private void showGroupNotificationForActiveNotificationsMap(int pushId, Bundle data) {
+        if (data == null)
+            return;
+
+        String largeIconUri = data.getString("icon");
+        String noteType = StringUtils.notNullStr(data.getString(PUSH_ARG_TYPE));
+        String noteId = data.getString(PUSH_ARG_NOTE_ID, "");
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: Should we return immediately  (and log a warning) if `noteId` is empty?

**Comment by mzorz** [comment]
Thanks for your comments @daniloercoli - most of them are covered in this new PR #4625 (and also the most important one, having fragments to be agnostic of all this is covered there as well). Will close this PR as we move onto reviewing #4625 if you agree 👍 🙇

**Comment by daniloercoli** [comment]
Closing this PR since the new one should fix the bug by using the approach we discussed in chat.

**Pull request closed without merging**